### PR TITLE
Test file of extensions

### PR DESCRIPTION
**Pull request by tychonievich**
The specific extension-related content is described in HEAD.NOTE. Note that not all of these cases are explicitly identified in the spec; notably, it includes

- extension tags aliasing standard tags
- relocated standard enumeration values, to parallel relocated standard structures
- relocated standard structure as a record, possibly at odds with this part from the spec (emphasis added) "These are called relocated standard structures and can only appear **with superstructures** that are not documented as a superstructure of that structure type in this specification"

Related to #87 -- if we like this file, we can use it to fill out maximal70.ged

### CODE REVIEW

**Comment by tychonievich** [comment]
On the aliasing tags (i.e. using `_USER` to mean `SUBM`, `_JOUR` to mean `COMP`, and so on): I think these violate the spirit of the [extension versus standard section](https://gedcom.io/specifications/FamilySearchGEDCOMv7.html#extension-versus-standard) of the specification, but I couldn't find any specific prohibition against them there. If we decide they are in violation of that, we should update that section to be explicit about that and remove them from this example file. One reason not to prohibit them, and thus to leave them in the example file, is the expectation that version 7.1 might make standard something that some 7.0 files include using extensions; if we fully ban aliasing that will violate the "every 7.0 file is a valid 7.1 file" principle.

**Comment by albertemmerich** [comment]
1. Yes, aliasing tags violate the spirit of extension versus standard section.
2. If we think the actual standard version does not explicitly prohibit aliasing tags, we cannot prohibit them in 7.x: In that case files which we think to be valid under actual spec 7.0 would be invalid after update.
3. There are very popular violations of this kind in the wild. See FTM and their _MREL and _FREL describing the kind of relation of a child to its father and mother, used in FAM records as subtags of tag CHIL. Standard requires to use PEDI in INDI record, and not any tag in FAM record. At import FTM converts adopted children as presented by PEDI in INDI records to biological children creating wrong _FREL and _MREL payloads. So this aliasing tags are cause of severe data transfer problems. Seeing this it would be very desirable to forbid these tags. However I think this is too late for 7.x.
4. I see another problem what "aliasing" really means - is it allowed to use an extension if you have a very small deviation in the definition? Saying the extension tag is very similar but not identical to a standard tag? If we want to find a definition what is forbidden and what is allowed I think we have to do some work - it isn't easy at all!
Summary: I think we are not ready to ban aliasing tags.

File: testfiles/gedcom70/extensions.ged [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+2 TAG _CHILD https://gedcom.io/terms/v7/enum-CHIL
+2 TAG _PHRASE https://gedcom.io/terms/v7/PHRASE
+2 TAG _PARTY http://example.com/party-participation
+2 TAG _PARTY http://example.com/party
+2 TAG _BIRT https://gedcom.io/terms/v7/BIRT
+1 NOTE This file contains the following extension-related content:
+2 CONT Standard record with an extTag           0 @U1@ _USER
+2 CONT Standard substructure                    1 NAME Aliased record
+2 CONT Standard structure with an extTag        1 _CREATOR @U1@
+2 CONT Standard structure with an extTag        1 _CREATOR @U1@
```
Comment (dthaler) [inline testfiles/gedcom70/extensions.ged:24]:
```suggestion
2 CONT Standard structure with an extTag        1 _CREATOR @U2@
```

File: testfiles/gedcom70/extensions.ged [hunk 2]
```diff
@@ -0,0 +1,93 @@
+0 HEAD
+1 GEDC
+2 VERS 7.0
+1 SCHMA
+2 TAG _RECORD http://example.com/record-type
+2 TAG _STRUCT http://example.com/substructure-type
+2 TAG _ENUMVAL http://example.com/enumeration-value
+2 TAG _CALENDAR http://example.com/calendar
+2 TAG _MONTH http://example.com/month
+2 TAG _EPOCH http://example.com/epoh
```
Comment (dthaler) [inline testfiles/gedcom70/extensions.ged:10]:
```suggestion
2 TAG _EPOCH http://example.com/epoch
```

File: testfiles/gedcom70/extensions.ged [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+2 CONT Undocumented structure with a pointer    1 _EXT1 @R1@
+2 CONT Undocumented pointer to relocated record 1 _IN @B1@
+2 CONT Unambiguous extension-defined substruct. 2 ROLE CHIL
+2 CONT Undocumented record                      0 @R1@ _RECORD not empty
+2 CONT Pointer to record with an extTag         1 SUBM @U1@
+2 CONT Standard enumeration with an extTag      2 ROLE _CHILD
+2 CONT Standard month with an extTag            ... FRENCH_R 2 _JOUR 8
+2 CONT Standard calendar with an extTag         ... _CALENDRIER 4 JOUR 8
+2 CONT Documented calendar, month, and epoch    ... _CALENDAR 8 _MONTH 190 _EPOCH
+2 CONT Undocumented calendar, month,  epoch     ... _CAL2 23 _MON2 88 _EP2
```
Comment (dthaler) [inline testfiles/gedcom70/extensions.ged:37]:
```suggestion
2 CONT Undocumented calendar, month, and epoch     ... _CAL2 23 _MON2 88 _EP2
```

File: testfiles/gedcom70/extensions.ged [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+2 CONT Documented record with shared tag        0 @P1@ _PARTY
+2 CONT Ambiguous extension-defined substructure 1 NAME Spring Fling
+2 CONT Undocumented record with shared tag      0 @L1@ _LOC
+2 CONT Extension-defined substructure           1 SUBM @S2@
+2 CONT Ambiguous extension-defined substructure 1 DATE TO 1880
+2 CONT Unambiguous nested ext.-def. substructu. 2 PHRASE Dissolved in the 1870s
+2 CONT Standard and relocated enumerations      2 EVEN DEAT, _CHILD
+2 CONT Relocated standard structure             3 _CREATOR @U2@
+2 CONT Substructure relocated as record         0 @B1@ _BIRT
+2 CONT Standard substructure of relocate struc. 1 PLAC Some, Place, , List
```
Comment (dthaler) [inline testfiles/gedcom70/extensions.ged:50]:
```suggestion
2 CONT Standard substructure of relocated stru. 1 PLAC Some, Place, , List
```

File: testfiles/gedcom70/extensions.ged [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+2 CONT Unambiguous extension-defined substruct. 2 ROLE CHIL
+2 CONT Undocumented record                      0 @R1@ _RECORD not empty
+2 CONT Pointer to record with an extTag         1 SUBM @U1@
+2 CONT Standard enumeration with an extTag      2 ROLE _CHILD
+2 CONT Standard month with an extTag            ... FRENCH_R 2 _JOUR 8
+2 CONT Standard calendar with an extTag         ... _CALENDRIER 4 JOUR 8
+2 CONT Documented calendar, month, and epoch    ... _CALENDAR 8 _MONTH 190 _EPOCH
+2 CONT Undocumented calendar, month,  epoch     ... _CAL2 23 _MON2 88 _EP2
+2 CONT Undocumented pointer with shared tag     2 _LOC @L1@
+2 CONT documented pointer with shared tag       1 _PARTY @P1@
```
Comment (dthaler) [inline testfiles/gedcom70/extensions.ged:39]:
```suggestion
2 CONT Documented pointer with shared tag       1 _PARTY @P1@
```
For consistency with line 41 etc

**Review by dthaler — COMMENTED** [review]
Per meeting discussion, please also add to maximal70.ged

**Comment by tychonievich** [comment]
To do:
- add to maximal.ged (changing identifiers to avoid collisions)
- link from tools page

File: testfiles/gedcom70/extensions.ged [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+2 TAG _EPOCH http://example.com/epoch
+2 TAG _USER https://gedcom.io/terms/v7/record-SUBM
+2 TAG _CREATOR https://gedcom.io/terms/v7/SUBM
+2 TAG _CALENDRIER https://gedcom.io/terms/v7/cal-FRENCH_R
+2 TAG _JOUR https://gedcom.io/terms/v7/month-COMP
+2 TAG _CHILD https://gedcom.io/terms/v7/enum-CHIL
+2 TAG _PHRASE https://gedcom.io/terms/v7/PHRASE
+2 TAG _PARTY http://example.com/party-participation
+2 TAG _PARTY http://example.com/party
+2 TAG _BIRT https://gedcom.io/terms/v7/BIRT
```
Comment (dthaler) [inline testfiles/gedcom70/extensions.ged:19]: should be a separate URI for a record, not reuse a non-record URI

Comment (dthaler) [reply testfiles/gedcom70/extensions.ged:19]: @tychonievich Bringing the above comment back to your attention.  Do you agree or disagree?

Comment (dthaler) [reply testfiles/gedcom70/extensions.ged:19]: Discussion March 7, 2024: just delete this line and the record.

**Review by tychonievich — COMMENTED** [review]
remove the illegal relocated record

**Review by dthaler — APPROVED** [review]

**Comment by dthaler** [comment]
Agreed in GEDCOM Steering Committee meeting April 11, 2024.
Separate issues will be filed for the todos:
1. add to maximal.ged (changing identifiers to avoid collisions)
2. link from tools page

**Pull request merged**